tasmota/xsns_88_tfminiplus.ino

```diff
@@ -0,0 +1,258 @@
+#define BAUDRATE 9600
+
+// Serial buffer length for incoming data
+#define TFMP_MAX_DATA_LEN 9
+
+char Tfmp_buffer[TFMP_MAX_DATA_LEN + 1];
+
+struct xsns_88_tfminiplus
+{
+    bool ready = false;
+    uint16_t distance = 0;
+    uint16_t sigstrength = 0;
+    uint16_t chiptemp = 0;
+} tfminiplus_sensor;
+
+// Software and hardware serial pointers
+#ifdef ESP8266
+TasmotaSerial *TfmpSerial = nullptr;
+#endif  // ESP8266
+#ifdef ESP32
+HardwareSerial *TfmpSerial = nullptr;
+#endif  // ESP32
+
+void TfmpInit(void)
+{
+    if (!tfminiplus_sensor.ready)
+    {
+        if (PinUsed(GPIO_TFMINIPLUS_RX) && PinUsed(GPIO_TFMINIPLUS_TX))
+        {
+#ifdef ESP8266
+            TfmpSerial = new TasmotaSerial(Pin(GPIO_TFMINIPLUS_RX), Pin(GPIO_TFMINIPLUS_TX), 1);
+            if (TfmpSerial->begin(BAUDRATE))
+            {
+                if (TfmpSerial->hardwareSerial())
+                {
+                    ClaimSerial();
+                }
+                tfminiplus_sensor.ready = true;
+                TfmpSerial->flush();
+            }
+#endif // ESP8266
+#ifdef ESP32
+            TfmpSerial = new HardwareSerial(2);
+            TfmpSerial->begin(BAUDRATE, SERIAL_8N1, Pin(GPIO_TFMINIPLUS_RX), Pin(GPIO_TFMINIPLUS_TX));
+            TfmpSerial->flush();
+#endif // ESP32
+        }
+    }
+}
+
+void TfmpTrigger(void)
+{
+    if (TfmpSerial && tfminiplus_sensor.ready)
+    {
+        TfmpSerial->write(0x5A);
+        TfmpSerial->write(0x04);
+        TfmpSerial->write(0x04);
+        TfmpSerial->write(0x62);
+    }
+}
+
+void TfmpProcessData(void)
+{
+    uint16_t crc = 0;
+    // distance to object (default in cm)
+    tfminiplus_sensor.distance = (Tfmp_buffer[3] << 8) + Tfmp_buffer[2];
+    // signal strength (between 100 and 65535 fine, otherwise 0 due to out of range of non-reflective surface)
+    tfminiplus_sensor.sigstrength = (Tfmp_buffer[5] << 8) + Tfmp_buffer[4];
+    // chip temperature
+    tfminiplus_sensor.chiptemp = (((Tfmp_buffer[7] << 8) + Tfmp_buffer[6]) >> 3) - 256;
+    // check crc sum
+    for (int i = 0; i < TFMP_MAX_DATA_LEN - 1; ++i)
+    {
+        crc += (uint16_t)Tfmp_buffer[i];
+    }
+    if (!(char)(crc & 0xff) != Tfmp_buffer[TFMP_MAX_DATA_LEN])
+    {
+        DEBUG_SENSOR_LOG(PSTR("TFmini Plus: crc error"));
+    }
+    DEBUG_SENSOR_LOG(PSTR("TFmini Plus: Distance: %d"), tfminiplus_sensor.distance);
+    DEBUG_SENSOR_LOG(PSTR("TFmini Plus: Signal: %d"), tfminiplus_sensor.sigstrength);
+    DEBUG_SENSOR_LOG(PSTR("TFmini Plus: Chip Temp: %d"), tfminiplus_sensor.chiptemp);
+    TfmpSerial->flush();
+}
+
+void TfmpProcessSerialData (void)
+{
+    uint8_t data;
+    bool dataReady;
+    if (TfmpSerial && tfminiplus_sensor.ready)
+    {
+        while (TfmpSerial->available() > 0) 
+        {
+            data = TfmpSerial->read();
+            dataReady = addData((char)data);
+            if (dataReady)
+            {
+                TfmpProcessData();
+            }
+        }
+    }
+}
+
+bool addData(char nextChar)
```

Comment:
Use Tfmp prefix for global function addData()

Reply:
Sure (was not consequent) and changed to Tfmp prefix. THX